### DISCUSSION TITLE
Problem with encoding ISO-8859-2

### DISCUSSION THREAD

**Discussion by meavric** [rappasoft/laravel-livewire-tables#1867] [General]
Hi,
I have problem, my dats in database are encoding in ISO-8859-2.
Now I have problem with popup model, I get error 404 when click in row with special chars ISO-8859-2.
Is there some posibbility to convert charset ISO-8859-2 geting from database to UTF-8 ?
Now I see bad chars in my dats rows.
Thank you for yours any help how fix it.
BR

**Comment by meavric** [comment]
Hi, 
I found other problem when row ID is big for example 64664 and I click on this row I get error 404 but if ID is low for example 31 then its work wine of course I have bad chars but model has opened.

I check and last ID when model is opened is 556, 557 and later are  not opened

**Reply by lrljoe** [reply]
Can you give me some examples of this please.  Some screenshots, stack traces, and Browser Console logs are always useful.

How are you loading/popping up your "popup model" (popup modal?)

What you've got set in your Eloquent Model in terms of the Primary Key

The Table is tested, and will work with UUID/ULID/Integer primary keys, I'd need to understand in more detail what the issue is to offer a fix.

**Comment by meavric** [comment]
Hi,
This is my code:
`<?php

namespace App\Livewire\Ur\Regeneration;

use App\Models\GcUrZgloszenia;
use App\Models\User;
use Rappasoft\LaravelLivewireTables\DataTableComponent;
use Rappasoft\LaravelLivewireTables\Views\Column;
use Rappasoft\LaravelLivewireTables\Views\Filter;
use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Facades\DB;
use Illuminate\Database\Eloquent\Collection;
use Rappasoft\LaravelLivewireTables\Views\Filters\SelectFilter;

class Regeneration extends DataTableComponent
{
    public $viewingHistoryModal, $currentHistory;

    public function configure(): void
    {
        $this->setDebugStatus(true);

        $this->setPrimaryKey('id')
        ->setTableRowUrl(fn () => '#')
        ->setTrAttributes(function ($row) {
            return ['wire:click.prevent' => 'viewHistoryModal('.$row->id.')'];
        });
    }

    public function builder(): Builder
    {
          $query = GcUrZgloszenia::query();
          return $query;
    }

    public function columns(): array
    {
        return [
            Column::make('ID', 'id')
                ->sortable(),
            Column::make('Name', 'komunikat')
                ->sortable(),
            Column::make('Status', 'status')
            ->sortable()
            ->searchable(),  // Wyszukiwanie w tej kolumnie

        ];
    }

    public function filters(): array
    {
        return [            
        SelectFilter::make('Status')
        ->setFilterDefaultValue('201')
        ->setFilterPillTitle('User Status')
        ->setFilterPillValues([
            '201' => 'Active 201',
            '206' => 'Inactive 206',
        ])
        ->options([
            '' => 'All',
            '206' => '206',
            '201' => '201',
        ])
        ->filter(function(Builder $builder, string $value) {
            if ($value === '206') {
                $builder->where('status', 206);
            } elseif ($value === '201') {
                $builder->where('status', 201);
            }
        }),];
    }

    public function viewHistoryModal($historyId): void
    {
        $this->viewingHistoryModal = true;
        $this->currentHistory = User::findOrFail($historyId);
        $this->dispatch('open-modal', 'kwalifikacjaModal');
    }

    public function customView(): string
    {
        // return 'livewire/ur/regeneration/regeneration';
        return 'includes/regeneration';
    }
}`




file: includes/regeneration.blade.php
`<div>
    <x-modal wire:model='currentHistory' name="kwalifikacjaModal" title="KWALIFIKACJA USZKODZENIA" :show="false"
        maxWidth="xl">
        <x-slot:body>
            <section class="bg-white dark:bg-gray-900">
                <div class="py-1 lg:py-16 px-4 mx-auto max-w-screen-2xl">

                    <div class="row">
                        @if(isset($kw_status))
                        <x-input-select wire:model.live='kw_status_save'>
                            <option selected required>{{_('Wybierz...')}}</option>
                            @foreach ($kw_status as $item)
                            <option value="{{$item->id}}">{{$item->name}}</option>
                            @endforeach
                        </x-input-select>
                        @endif
                    </div>

                    <div class="modal-footer">
                        @if(!empty($kw_status_save))
                        <x-button-save wire:click="kw_save('kwalifikacjaModal','{{$regenerationId}}')">ZAPISZ
                        </x-button-save>
                        @endif
                        <x-button-close wire:click="close('kwalifikacjaModal')">ZAMKNIJ</x-button-show>
                    </div>

                </div>
            </section>
        </x-slot:body>

    </x-modal>
</div>`


I attach 3 ss:
- row_id_64666 - when error 404
- error_404
- ok - when work fine

![row_id_64666](https://github.com/user-attachments/assets/63f7f04b-146b-499c-81bd-4e3d923be95e)
![ok](https://github.com/user-attachments/assets/9a0fe4ca-d589-4ffd-b28b-228b867065c9)
![error_404](https://github.com/user-attachments/assets/870e5231-a7cf-4102-92cf-432a713ebe5b)


In row id 37 you can see bad char - how can I fix it?

I add that line with error: livewire.js?id=87e1046f:4275 -> say: response = await fetch(updateUri, options);

I change primaryKey to other column but problem is the same.

I change query to:
`          $query = GcUrZgloszenia::query();
          $query->where('id', '>','500')->where('id', '<','600');`
but problem still exist


Thank you for your help
BR

**Comment by meavric** [comment]
Hi,
Any idea what can be wrong?
Maybe some idea how fix charset?
BR

**Reply by lrljoe** [reply]
Right, you're calling:
```
public function viewHistoryModal($historyId): void
```
And passing it "historyId" (i.e. the Primary Key for the row of GcUrZgloszenia)

**HOWEVER**

BUT

You're then expecting the historyId to match a User primary key, which won't always be the case, unless it's a one-to-one relation, with matching keys.

I suspect you need to get the "user_id" for the GcUrZgloszenia and pass that into the relevant methods.

Otherwise given that it's given you a 404 for the livewire.js file, that's not the Tables, it's something else!  One or more of your screenshots are returning a 404 error.  Can you possibly do a screen capture/video capture, which may help clarify.

You've also not shared the full code for
includes/regeneration.blade.php

Which makes it harder to diagnose further.

**Comment by meavric** [comment]
Hi
Thank you for your answer.

I don't know why but I changed table with datas to other and go back to GcUrZgloszenia and start working fine.
I dont know why, nothing more I don't changed.

Plase help me resolve my problem with PL locale - encoding ISO-8859-2. How make conversation from ISO-8859-2 to UTF-8 to show fix text in main table?

Now I use:
            Column::make('Opis', 'komunikat')
                ->sortable()
                ->searchable()
                ->format(
                    fn($value, $row, Column $column) => iconv('Windows-1250', 'UTF-8//IGNORE', $row->komunikat)
                ),

BR
